File: sdks/php/src/Models/PercentageRolloutEntry.php
<?php

declare(strict_types=1);

namespace PhaseFlag\Models;

/**
 * One slice of a percentage rollout.
 *
 * Pairs a variation ID with a weight (0-100) that determines what proportion
 * of traffic is assigned to this variation.
 */
final class PercentageRolloutEntry
{
    public function __construct(
        public readonly string $variationId,
        public readonly int $weight,
    ) {}

    /**
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        return [
            'variation_id' => $this->variationId,
            'weight'       => $this->weight,
        ];
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): static
    {
        return new static(
            variationId: $data['variation_id'] ?? '',
            weight: (int) ($data['weight'] ?? 0),
        );
    }
}

File: sdks/dotnet/src/Models/PercentageRolloutEntry.cs
using System.Text.Json.Serialization;

namespace PhaseFlag.Models;

/// <summary>
/// One slice of a percentage rollout, mapping a variation to a weight.
/// </summary>
public sealed class PercentageRolloutEntry
{
    /// <summary>The variation identifier for this rollout slice.</summary>
    [JsonPropertyName("variation_id")]
    public string VariationId { get; set; } = string.Empty;

    /// <summary>
    /// The weight of this slice as an integer percentage point (0-100).
    /// The sum of all weights in a rollout should equal 100.
    /// </summary>
    [JsonPropertyName("weight")]
    public int Weight { get; set; }
}
